service/src/main/kotlin/fi/espoo/evaka/vasu/VasuController.kt

```diff
@@ -0,0 +1,91 @@
+package fi.espoo.evaka.vasu
+
+import fi.espoo.evaka.shared.auth.AuthenticatedUser
+import fi.espoo.evaka.shared.auth.UserRole
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.domain.BadRequest
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import java.net.URI
+import java.time.LocalDate
+import java.util.UUID
+
+@RestController
+@RequestMapping("/vasu")
+class VasuController {
+
+    data class CreateDocumentRequest(
+        val childId: UUID,
+        val templateId: UUID
+    )
+    @PostMapping
+    fun createDocument(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @RequestBody body: CreateDocumentRequest
+    ): ResponseEntity<VasuDocumentResponse> {
+        user.requireOneOfRoles(UserRole.ADMIN)
+
+        val document = db.transaction { tx ->
+            tx.getVasuTemplate(body.templateId).let { template ->
+                if (!template.valid.includes(LocalDate.now())) {
+                    throw BadRequest("Template is not currently valid")
+                }
+            }
+            val id = tx.insertVasuDocument(body.childId, body.templateId)
+            tx.getVasuDocumentResponse(id)
+        }
+
+        return ResponseEntity.created(URI.create("/vasu/${document.id}")).body(document)
+    }
+
+    @GetMapping("/{id}")
+    fun getDocument(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @PathVariable id: UUID
+    ): ResponseEntity<VasuDocumentResponse> {
+        user.requireOneOfRoles(UserRole.ADMIN)
+
+        return db.read { tx ->
+            tx.getVasuDocumentResponse(id)
+        }.let { ResponseEntity.ok(it) }
```

Comment:
You can just use `VasuDocumentResponse` as return type and return the transaction result `return db.read { it -> getVasuDocumentResponse(id) }`. Same goes for other `ResponseEntity` usages.